nestkernel/parameter.h

```diff
@@ -1724,6 +1724,275 @@ class DimensionParameter : public Parameter
+  {
+    parameter_is_spatial_ = true;
+  }
+
+  ~ExpDistParameter() override
+  {
+    delete p_;
+  }
+
+  /**
+   * @returns the value of the parameter.
+   */
+  double
+  value( RngPtr, Node* ) override
+  {
+    throw BadParameterValue( "Exponential distribution parameter can only be used when connecting." );
+  }
+
+  double
+  value( RngPtr, index, Node*, thread ) override
+  {
+    throw KernelException( "Exponential distribution parameter can only be used when using ConnectLayers." );
+  }
+  double value( RngPtr rng,
+    const std::vector< double >& source_pos,
+    const std::vector< double >& target_pos,
+    const AbstractLayer& layer ) override;
+
+  Parameter*
+  clone() const override
+  {
+    return new ExpDistParameter( *this );
+  }
+
+protected:
+  Parameter* p_;
+  const double beta_;
+};
+
+
+/**
+ * Parameter class representing a gaussian distribution applied on a parameter.
+ */
+class GaussianParameter : public Parameter
+{
+public:
+  using Parameter::value;
+
+  /**
+   * Construct the parameter from a dictionary of arguments.
+   */
+  GaussianParameter( const DictionaryDatum& d );
+
+  /**
+   * Copy constructor.
+   */
+  GaussianParameter( const GaussianParameter& p )
+    : Parameter( p )
+    , p_( p.p_->clone() )
+    , mean_( p.mean_ )
+    , std_( p.std_ )
+    , two_std2_( p.two_std2_ )
+  {
+    parameter_is_spatial_ = true;
+  }
+
+  ~GaussianParameter() override
+  {
+    delete p_;
+  }
+
+  /**
+   * @returns the value of the parameter.
+   */
+  double
+  value( RngPtr, Node* ) override
+  {
+    throw BadParameterValue( "Gaussian distribution parameter can only be used when connecting." );
+  }
+
+  double
+  value( RngPtr, index, Node*, thread ) override
+  {
+    throw KernelException( "Gaussian distribution parameter can only be used when using ConnectLayers." );
+  }
+  double value( RngPtr rng,
+    const std::vector< double >& source_pos,
+    const std::vector< double >& target_pos,
+    const AbstractLayer& layer ) override;
+
+  Parameter*
+  clone() const override
+  {
+    return new GaussianParameter( *this );
+  }
+
+protected:
+  Parameter* p_;
+  const double mean_;
+  const double std_;
+  const double two_std2_;
+};
+
+
+/**
+ * Parameter class representing a gaussian distribution in two dimensions applied on a parameter.
+ */
+class Gaussian2DParameter : public Parameter
+{
+public:
+  using Parameter::value;
+
+  /**
+   * Construct the parameter from a dictionary of arguments.
+   */
+  Gaussian2DParameter( const DictionaryDatum& d );
+
+  /**
+   * Copy constructor.
+   */
+  Gaussian2DParameter( const Gaussian2DParameter& p )
+    : Parameter( p )
+    , px_( p.px_->clone() )
+    , py_( p.py_->clone() )
+    , mean_x_( p.mean_x_ )
+    , mean_y_( p.mean_y_ )
+    , std_x_( p.std_x_ )
+    , std_y_( p.std_y_ )
+    , rho_( p.rho_ )
+    , std_x2_( p.std_x2_ )
+    , std_y2_( p.std_y2_ )
+    , std_xy_( p.std_xy_ )
+    , two_sub_2rho2_( p.two_sub_2rho2_ )
```

Comment:
Do we need to keep all parameters around, e.g., `std_x_`, `std_y_`, `rho_`? It should be enough to keep just the precomputed values we actually use in the calculations, shouldn't it? Could also apply to other of these parameters.